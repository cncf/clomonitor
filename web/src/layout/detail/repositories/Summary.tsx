import { FaCrown } from 'react-icons/fa';
import { VscGithub } from 'react-icons/vsc';
import { useLocation, useNavigate } from 'react-router-dom';

import { CATEGORY_ICONS } from '../../../data';
import { Repository, RepositoryKind, ScoreType } from '../../../types';
import ElementWithTooltip from '../../common/ElementWithTooltip';
import Badge from './Badge';
import styles from './Summary.module.css';

interface Props {
  repositories: Repository[];
}

const Summary = (props: Props) => {
  const navigate = useNavigate();
  const location = useLocation();

  if (props.repositories.length === 0) return null;
  return (
    <div className="pt-2 mb-4 mb-md-5 table-responsive">
      <table className={`table table-bordered mb-0 ${styles.table}`}>
        <thead>
          <tr>
            <th scope="col" className="text-center text-nowrap">
              <small className={`me-2 position-relative ${styles.icon}`}>
                <VscGithub />
              </small>
              <span>Repository</span>
            </th>
            <th scope="col" className="text-center text-nowrap">
              <small className={`me-2 position-relative ${styles.icon}`}>{CATEGORY_ICONS[ScoreType.Global]}</small>
              <span>Global</span>
            </th>
            <th scope="col" className="d-none d-md-table-cell text-center text-nowrap">
              <small className={`me-2 position-relative ${styles.icon}`}>
                {CATEGORY_ICONS[ScoreType.Documentation]}
              </small>
              <span>Documentation</span>
            </th>
            <th scope="col" className="d-none d-md-table-cell text-center text-nowrap">
              <small className={`me-2 position-relative ${styles.icon}`}>{CATEGORY_ICONS[ScoreType.License]}</small>
              <span>License</span>
            </th>
            <th scope="col" className="d-none d-md-table-cell text-center text-nowrap">
              <small className={`me-2 position-relative ${styles.icon}`}>
                {CATEGORY_ICONS[ScoreType.BestPractices]}
              </small>
              <span>Best Practices</span>
            </th>
            <th scope="col" className="d-none d-md-table-cell text-center text-nowrap">
              <small className={`me-2 position-relative ${styles.icon}`}>{CATEGORY_ICONS[ScoreType.Security]}</small>
              <span>Security</span>
            </th>
          </tr>
        </thead>
        <tbody>
          {props.repositories.map((repo: Repository) => {
            return (
              <tr key={`summary_${repo.repositoryId}`}>
                <td className={`align-middle ${styles.repoCell} ${styles.darkBgCell}`}>
                  <div className="d-flex flex-row align-items-center">
                    <button
                      className={`btn btn-link text-dark text-truncate fw-bold ${styles.repoBtn}`}
                      onClick={() =>
                        navigate(
                          {
                            pathname: location.pathname,
                            hash: repo.name,
                          },
                          { state: location.state }
                        )
                      }
                    >
                      {repo.name}
                    </button>
                    {repo.kind === RepositoryKind.Primary && (
                      <>
                        <FaCrown className="d-block d-md-none text-warning" />
                        <ElementWithTooltip
                          className="lh-1"
                          element={<FaCrown className="text-warning" />}
                          tooltipWidth={210}
                          tooltipClassName={styles.tooltipMessage}
                          tooltipMessage={<div>Primary project's repository</div>}
                          alignmentTooltip="left"
                          forceAlignment
                          visibleTooltip
                          active
                        />
                      </>
                    )}
                  </div>
                </td>
                <td className="align-middle">
                  <Badge value={repo.score.global} />
                </td>
                <td className="d-none d-md-table-cell align-middle">
                  <Badge value={repo.score.documentation} />
                </td>
                <td className="d-none d-md-table-cell align-middle">
                  <Badge value={repo.score.license} />
                </td>
                <td className="d-none d-md-table-cell align-middle">
                  <Badge value={repo.score.bestPractices} />
                </td>
                <td className="d-none d-md-table-cell align-middle">
                  <Badge value={repo.score.security} />
                </td>
              </tr>
            );
          })}
        </tbody>
      </table>
    </div>
  );
};

export default Summary;
