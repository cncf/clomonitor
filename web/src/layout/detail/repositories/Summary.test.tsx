import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { BrowserRouter as Router } from 'react-router-dom';

import { Repository } from '../../../types';
import Summary from './Summary';

const mockUseNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...(jest.requireActual('react-router-dom') as any),
  useNavigate: () => mockUseNavigate,
}));

const getRepositories = (fixtureId: string): Repository[] => {
  return require(`./__fixtures__/Summary/${fixtureId}.json`) as Repository[];
};

const mockScrollIntoView = jest.fn();

const defaultProps = {
  scrollIntoView: mockScrollIntoView,
};

describe('Summary', () => {
  afterEach(() => {
    jest.resetAllMocks();
  });

  it('creates snapshot', () => {
    const repositories = getRepositories('1');
    const { asFragment } = render(
      <Router>
        <Summary {...defaultProps} repositories={repositories} />
      </Router>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  describe('Render', () => {
    it('renders component', () => {
      const repositories = getRepositories('1');
      render(
        <Router>
          <Summary {...defaultProps} repositories={repositories} />
        </Router>
      );

      expect(screen.getByText('Repository')).toBeInTheDocument();
      expect(screen.getByText('Global')).toBeInTheDocument();
      expect(screen.getByText('Documentation')).toBeInTheDocument();
      expect(screen.getByText('License')).toBeInTheDocument();
      expect(screen.getByText('Best Practices')).toBeInTheDocument();
      expect(screen.getByText('Security')).toBeInTheDocument();

      expect(screen.getByText('spec')).toBeInTheDocument();
      expect(screen.getByText('sdk-go')).toBeInTheDocument();
      expect(screen.getByText('sdk-javascript')).toBeInTheDocument();
      expect(screen.getByText('sdk-csharp')).toBeInTheDocument();
      expect(screen.getByText('sdk-java')).toBeInTheDocument();
      expect(screen.getByText('sdk-python')).toBeInTheDocument();

      const btns = screen.getAllByRole('button', { name: /Go from summary to section:/i });
      expect(btns).toHaveLength(6);

      expect(screen.getByText('44')).toBeInTheDocument();
      expect(screen.getByText('60')).toBeInTheDocument();
      expect(screen.getByText('80')).toBeInTheDocument();
      expect(screen.getByText('35')).toBeInTheDocument();
      expect(screen.getByText('0')).toBeInTheDocument();

      expect(screen.getAllByText('95')).toHaveLength(2);
      expect(screen.getAllByText('90')).toHaveLength(2);
      expect(screen.getAllByText('100')).toHaveLength(5);
      expect(screen.getAllByText('85')).toHaveLength(3);
      expect(screen.getAllByText('70')).toHaveLength(3);
      expect(screen.getAllByText('n/a')).toHaveLength(16);
    });
  });

  it('renders component when one of the repos failed', () => {
    const repositories = getRepositories('2');
    render(
      <Router>
        <Summary {...defaultProps} repositories={repositories} />
      </Router>
    );

    expect(screen.getByText('Repository')).toBeInTheDocument();
    expect(screen.getByText('Global')).toBeInTheDocument();
    expect(screen.getByText('Documentation')).toBeInTheDocument();
    expect(screen.getByText('License')).toBeInTheDocument();
    expect(screen.getByText('Best Practices')).toBeInTheDocument();
    expect(screen.getByText('Security')).toBeInTheDocument();

    expect(screen.getByText('go-control-panel')).toBeInTheDocument();
    expect(screen.getByText('envoy')).toBeInTheDocument();

    const btns = screen.getAllByRole('button', { name: /Go from summary to section:/i });
    expect(btns).toHaveLength(2);

    expect(screen.getByText('75')).toBeInTheDocument();
    expect(screen.getByText('100')).toBeInTheDocument();

    expect(screen.getAllByText('90')).toHaveLength(2);
    expect(screen.getAllByText('87')).toHaveLength(2);
    expect(screen.getAllByText('n/a')).toHaveLength(6);
  });

  it('renders component', async () => {
    const repositories = getRepositories('1');
    render(
      <Router>
        <Summary {...defaultProps} repositories={repositories} />
      </Router>
    );

    const btn = screen.getAllByRole('button', { name: 'Go from summary to section: spec' });
    await userEvent.click(btn[0]);

    expect(mockUseNavigate).toHaveBeenCalledTimes(1);
    expect(mockUseNavigate).toHaveBeenCalledWith(
      {
        hash: 'spec',
        pathname: '/',
      },
      { state: null }
    );
  });

  it('does not render repository without report', () => {
    const repositories = getRepositories('3');
    render(
      <Router>
        <Summary {...defaultProps} repositories={repositories} />
      </Router>
    );

    expect(screen.queryByText('grpc.io')).toBeNull();
  });

  describe('Does not render', () => {
    it('when repositories is empty', () => {
      const { container } = render(
        <Router>
          <Summary {...defaultProps} repositories={[]} />
        </Router>
      );
      expect(container).toBeEmptyDOMElement();
    });
  });
});
