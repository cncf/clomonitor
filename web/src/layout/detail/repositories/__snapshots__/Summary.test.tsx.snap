// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Summary creates snapshot 1`] = `
<DocumentFragment>
  <div
    class="pt-2 mb-4 mb-md-5"
  >
    <table
      class="table table-bordered mb-0 w-100 table"
      data-testid="repositories-summary"
    >
      <thead>
        <tr>
          <th
            class="text-center text-nowrap"
            scope="col"
          >
            <small
              class="me-2 position-relative icon"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12 0a12 12 0 1 0 0 24 12 12 0 0 0 0-24zm3.163 21.783h-.093a.513.513 0 0 1-.382-.14.513.513 0 0 1-.14-.372v-1.406c.006-.467.01-.94.01-1.416a3.693 3.693 0 0 0-.151-1.028 1.832 1.832 0 0 0-.542-.875 8.014 8.014 0 0 0 2.038-.471 4.051 4.051 0 0 0 1.466-.964c.407-.427.71-.943.885-1.506a6.77 6.77 0 0 0 .3-2.13 4.138 4.138 0 0 0-.26-1.476 3.892 3.892 0 0 0-.795-1.284 2.81 2.81 0 0 0 .162-.582c.033-.2.05-.402.05-.604 0-.26-.03-.52-.09-.773a5.309 5.309 0 0 0-.221-.763.293.293 0 0 0-.111-.02h-.11c-.23.002-.456.04-.674.111a5.34 5.34 0 0 0-.703.26 6.503 6.503 0 0 0-.661.343c-.215.127-.405.249-.573.362a9.578 9.578 0 0 0-5.143 0 13.507 13.507 0 0 0-.572-.362 6.022 6.022 0 0 0-.672-.342 4.516 4.516 0 0 0-.705-.261 2.203 2.203 0 0 0-.662-.111h-.11a.29.29 0 0 0-.11.02 5.844 5.844 0 0 0-.23.763c-.054.254-.08.513-.081.773 0 .202.017.404.051.604.033.199.086.394.16.582A3.888 3.888 0 0 0 5.702 10a4.142 4.142 0 0 0-.263 1.476 6.871 6.871 0 0 0 .292 2.12c.181.563.483 1.08.884 1.516.415.422.915.75 1.466.964.653.25 1.337.41 2.033.476a1.828 1.828 0 0 0-.452.633 2.99 2.99 0 0 0-.2.744 2.754 2.754 0 0 1-1.175.27 1.788 1.788 0 0 1-1.065-.3 2.904 2.904 0 0 1-.752-.824 3.1 3.1 0 0 0-.292-.382 2.693 2.693 0 0 0-.372-.343 1.841 1.841 0 0 0-.432-.24 1.2 1.2 0 0 0-.481-.101c-.04.001-.08.005-.12.01a.649.649 0 0 0-.162.02.408.408 0 0 0-.13.06.116.116 0 0 0-.06.1.33.33 0 0 0 .14.242c.093.074.17.131.232.171l.03.021c.133.103.261.214.382.333.112.098.213.209.3.33.09.119.168.246.231.381.073.134.15.288.231.463.188.474.522.875.954 1.145.453.243.961.364 1.476.351.174 0 .349-.01.522-.03.172-.028.343-.057.515-.091v1.743a.5.5 0 0 1-.533.521h-.062a10.286 10.286 0 1 1 6.324 0v.005z"
                />
              </svg>
            </small>
            <span>
              Repository
            </span>
          </th>
          <th
            class="text-center text-nowrap"
            scope="col"
          >
            <small
              class="position-relative icon"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M21 4h-3V3a1 1 0 0 0-1-1H7a1 1 0 0 0-1 1v1H3a1 1 0 0 0-1 1v3c0 4.31 1.799 6.91 4.819 7.012A6.001 6.001 0 0 0 11 17.91V20H9v2h6v-2h-2v-2.09a6.01 6.01 0 0 0 4.181-2.898C20.201 14.91 22 12.31 22 8V5a1 1 0 0 0-1-1zM4 8V6h2v6.83C4.216 12.078 4 9.299 4 8zm8 8c-2.206 0-4-1.794-4-4V4h8v8c0 2.206-1.794 4-4 4zm6-3.17V6h2v2c0 1.299-.216 4.078-2 4.83z"
                />
              </svg>
            </small>
            <span
              class="d-inline-block d-md-none d-xl-inline-block ms-1 ms-xl-2"
            >
              Global
            </span>
          </th>
          <th
            class="d-none d-md-table-cell text-center text-nowrap"
            scope="col"
          >
            <small
              class="position-relative icon"
            >
              <svg
                fill="none"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M11 5H6a2 2 0 00-2 2v11a2 2 0 002 2h11a2 2 0 002-2v-5m-1.414-9.414a2 2 0 112.828 2.828L11.828 15H9v-2.828l8.586-8.586z"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </svg>
            </small>
            <span
              class="d-none d-xl-inline-block ms-1 ms-xl-2"
            >
              Documentation
            </span>
          </th>
          <th
            class="d-none d-md-table-cell text-center text-nowrap"
            scope="col"
          >
            <small
              class="position-relative icon"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M256 32c-70.7 0-128 57.3-128 128s57.3 128 128 128 128-57.3 128-128S326.7 32 256 32zm0 208c-44.2 0-80-35.8-80-80s35.8-80 80-80 80 35.8 80 80-35.8 80-80 80zM193.7 307.4c-19.1-8.1-36.2-19.6-50.8-34.3-1.4-1.4-2.8-2.8-4.1-4.3L64 400h96l48 80 48-105.8 25.5-56.2c-8.4 1.3-16.9 2-25.5 2-21.6 0-42.5-4.2-62.3-12.6zM373.3 268.9c-1.3 1.4-2.7 2.9-4.1 4.3-14.6 14.6-31.7 26.2-50.7 34.2L294 361.2l-21.9 48.4L304 480l48-80h96l-74.7-131.1z"
                />
              </svg>
            </small>
            <span
              class="d-none d-xl-inline-block ms-1 ms-xl-2"
            >
              License
            </span>
          </th>
          <th
            class="d-none d-md-table-cell text-center text-nowrap"
            scope="col"
          >
            <small
              class="position-relative icon"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g>
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                  <path
                    d="M5 4.604v9.185a4 4 0 0 0 1.781 3.328L12 20.597l5.219-3.48A4 4 0 0 0 19 13.79V4.604L12 3.05 5 4.604zM3.783 2.826L12 1l8.217 1.826a1 1 0 0 1 .783.976v9.987a6 6 0 0 1-2.672 4.992L12 23l-6.328-4.219A6 6 0 0 1 3 13.79V3.802a1 1 0 0 1 .783-.976zM12 13.5l-2.939 1.545.561-3.272-2.377-2.318 3.286-.478L12 6l1.47 2.977 3.285.478-2.377 2.318.56 3.272L12 13.5z"
                  />
                </g>
              </svg>
            </small>
            <span
              class="d-none d-xl-inline-block ms-1 ms-xl-2"
            >
              Best Practices
            </span>
          </th>
          <th
            class="d-none d-md-table-cell text-center text-nowrap"
            scope="col"
          >
            <small
              class="position-relative icon"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12 2C9.243 2 7 4.243 7 7v2H6c-1.103 0-2 .897-2 2v9c0 1.103.897 2 2 2h12c1.103 0 2-.897 2-2v-9c0-1.103-.897-2-2-2h-1V7c0-2.757-2.243-5-5-5zM9 7c0-1.654 1.346-3 3-3s3 1.346 3 3v2H9V7zm9.002 13H13v-2.278c.595-.347 1-.985 1-1.722 0-1.103-.897-2-2-2s-2 .897-2 2c0 .736.405 1.375 1 1.722V20H6v-9h12l.002 9z"
                />
              </svg>
            </small>
            <span
              class="d-none d-xl-inline-block ms-1 ms-xl-2"
            >
              Security
            </span>
          </th>
          <th
            class="d-none d-md-table-cell text-center text-nowrap"
            scope="col"
          >
            <small
              class="position-relative icon"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 14 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M7 4c-.83 0-1.5-.67-1.5-1.5S6.17 1 7 1s1.5.67 1.5 1.5S7.83 4 7 4zm7 6c0 1.11-.89 2-2 2h-1c-1.11 0-2-.89-2-2l2-4h-1c-.55 0-1-.45-1-1H8v8c.42 0 1 .45 1 1h1c.42 0 1 .45 1 1H3c0-.55.58-1 1-1h1c0-.55.58-1 1-1h.03L6 5H5c0 .55-.45 1-1 1H3l2 4c0 1.11-.89 2-2 2H2c-1.11 0-2-.89-2-2l2-4H1V5h3c0-.55.45-1 1-1h4c.55 0 1 .45 1 1h3v1h-1l2 4zM2.5 7L1 10h3L2.5 7zM13 10l-1.5-3-1.5 3h3z"
                  fill-rule="evenodd"
                />
              </svg>
            </small>
            <span
              class="d-none d-xl-inline-block ms-1 ms-xl-2"
            >
              Legal
            </span>
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td
            class="align-middle repoCell darkBgCell"
          >
            <div
              class="d-flex flex-row align-items-center"
            >
              <button
                aria-label="Go from summary to section: sdk-go"
                class="btn btn-link text-dark text-truncate fw-bold px-2 py-0 py-xl-1 repoBtn"
              >
                sdk-go
              </button>
              <div
                class="d-inline-flex d-none d-xl-inline-flex"
              >
                <div
                  class="badge text-secondary border border-secondary rounded-0 position-relative text-uppercase miniBadge d-none d-xl-inline-flex"
                  data-testid="repo-kind-badge"
                >
                  <div
                    class="d-flex flex-row align-items-center"
                  >
                    code
                  </div>
                </div>
              </div>
            </div>
            <div
              class="d-inline-flex d-inline-flex d-xl-none px-2"
            >
              <div
                class="badge text-secondary border border-secondary rounded-0 position-relative text-uppercase miniBadge d-inline-flex d-xl-none px-2"
                data-testid="repo-kind-badge"
              >
                <div
                  class="d-flex flex-row align-items-center"
                >
                  code
                </div>
              </div>
            </div>
          </td>
          <td
            class="hoverableCell align-middle"
            role="button"
          >
            <div
              class="mx-auto px-2 text-center text-dark fw-bold badge"
            >
              95
            </div>
          </td>
          <td
            class="hoverableCell d-none d-md-table-cell align-middle"
            role="button"
          >
            <div
              class="mx-auto px-2 text-center text-dark fw-bold badge"
            >
              90
            </div>
          </td>
          <td
            class="hoverableCell d-none d-md-table-cell align-middle"
            role="button"
          >
            <div
              class="mx-auto px-2 text-center text-dark fw-bold badge"
            >
              100
            </div>
          </td>
          <td
            class="d-none d-md-table-cell align-middle"
          >
            <div
              class="mx-auto px-2 text-center"
            >
              <span
                class="text-muted"
              >
                n/a
              </span>
            </div>
          </td>
          <td
            class="d-none d-md-table-cell align-middle"
          >
            <div
              class="mx-auto px-2 text-center"
            >
              <span
                class="text-muted"
              >
                n/a
              </span>
            </div>
          </td>
          <td
            class="d-none d-md-table-cell align-middle"
          >
            <div
              class="mx-auto px-2 text-center"
            >
              <span
                class="text-muted"
              >
                n/a
              </span>
            </div>
          </td>
        </tr>
        <tr>
          <td
            class="align-middle repoCell darkBgCell"
          >
            <div
              class="d-flex flex-row align-items-center"
            >
              <button
                aria-label="Go from summary to section: sdk-javascript"
                class="btn btn-link text-dark text-truncate fw-bold px-2 py-0 py-xl-1 repoBtn"
              >
                sdk-javascript
              </button>
              <div
                class="d-inline-flex d-none d-xl-inline-flex"
              >
                <div
                  class="badge text-secondary border border-secondary rounded-0 position-relative text-uppercase miniBadge d-none d-xl-inline-flex"
                  data-testid="repo-kind-badge"
                >
                  <div
                    class="d-flex flex-row align-items-center"
                  >
                    code
                  </div>
                </div>
              </div>
            </div>
            <div
              class="d-inline-flex d-inline-flex d-xl-none px-2"
            >
              <div
                class="badge text-secondary border border-secondary rounded-0 position-relative text-uppercase miniBadge d-inline-flex d-xl-none px-2"
                data-testid="repo-kind-badge"
              >
                <div
                  class="d-flex flex-row align-items-center"
                >
                  code
                </div>
              </div>
            </div>
          </td>
          <td
            class="hoverableCell align-middle"
            role="button"
          >
            <div
              class="mx-auto px-2 text-center text-dark fw-bold badge"
            >
              95
            </div>
          </td>
          <td
            class="hoverableCell d-none d-md-table-cell align-middle"
            role="button"
          >
            <div
              class="mx-auto px-2 text-center text-dark fw-bold badge"
            >
              90
            </div>
          </td>
          <td
            class="hoverableCell d-none d-md-table-cell align-middle"
            role="button"
          >
            <div
              class="mx-auto px-2 text-center text-dark fw-bold badge"
            >
              100
            </div>
          </td>
          <td
            class="d-none d-md-table-cell align-middle"
          >
            <div
              class="mx-auto px-2 text-center"
            >
              <span
                class="text-muted"
              >
                n/a
              </span>
            </div>
          </td>
          <td
            class="d-none d-md-table-cell align-middle"
          >
            <div
              class="mx-auto px-2 text-center"
            >
              <span
                class="text-muted"
              >
                n/a
              </span>
            </div>
          </td>
          <td
            class="d-none d-md-table-cell align-middle"
          >
            <div
              class="mx-auto px-2 text-center"
            >
              <span
                class="text-muted"
              >
                n/a
              </span>
            </div>
          </td>
        </tr>
        <tr>
          <td
            class="align-middle repoCell darkBgCell"
          >
            <div
              class="d-flex flex-row align-items-center"
            >
              <button
                aria-label="Go from summary to section: sdk-csharp"
                class="btn btn-link text-dark text-truncate fw-bold px-2 py-0 py-xl-1 repoBtn"
              >
                sdk-csharp
              </button>
              <div
                class="d-inline-flex d-none d-xl-inline-flex"
              >
                <div
                  class="badge text-secondary border border-secondary rounded-0 position-relative text-uppercase miniBadge d-none d-xl-inline-flex"
                  data-testid="repo-kind-badge"
                >
                  <div
                    class="d-flex flex-row align-items-center"
                  >
                    code
                  </div>
                </div>
              </div>
            </div>
            <div
              class="d-inline-flex d-inline-flex d-xl-none px-2"
            >
              <div
                class="badge text-secondary border border-secondary rounded-0 position-relative text-uppercase miniBadge d-inline-flex d-xl-none px-2"
                data-testid="repo-kind-badge"
              >
                <div
                  class="d-flex flex-row align-items-center"
                >
                  code
                </div>
              </div>
            </div>
          </td>
          <td
            class="hoverableCell align-middle"
            role="button"
          >
            <div
              class="mx-auto px-2 text-center text-dark fw-bold badge"
            >
              85
            </div>
          </td>
          <td
            class="hoverableCell d-none d-md-table-cell align-middle"
            role="button"
          >
            <div
              class="mx-auto px-2 text-center text-dark fw-bold badge"
            >
              70
            </div>
          </td>
          <td
            class="hoverableCell d-none d-md-table-cell align-middle"
            role="button"
          >
            <div
              class="mx-auto px-2 text-center text-dark fw-bold badge"
            >
              100
            </div>
          </td>
          <td
            class="d-none d-md-table-cell align-middle"
          >
            <div
              class="mx-auto px-2 text-center"
            >
              <span
                class="text-muted"
              >
                n/a
              </span>
            </div>
          </td>
          <td
            class="d-none d-md-table-cell align-middle"
          >
            <div
              class="mx-auto px-2 text-center"
            >
              <span
                class="text-muted"
              >
                n/a
              </span>
            </div>
          </td>
          <td
            class="d-none d-md-table-cell align-middle"
          >
            <div
              class="mx-auto px-2 text-center"
            >
              <span
                class="text-muted"
              >
                n/a
              </span>
            </div>
          </td>
        </tr>
        <tr>
          <td
            class="align-middle repoCell darkBgCell"
          >
            <div
              class="d-flex flex-row align-items-center"
            >
              <button
                aria-label="Go from summary to section: sdk-python"
                class="btn btn-link text-dark text-truncate fw-bold px-2 py-0 py-xl-1 repoBtn"
              >
                sdk-python
              </button>
              <div
                class="d-inline-flex d-none d-xl-inline-flex"
              >
                <div
                  class="badge text-secondary border border-secondary rounded-0 position-relative text-uppercase miniBadge d-none d-xl-inline-flex"
                  data-testid="repo-kind-badge"
                >
                  <div
                    class="d-flex flex-row align-items-center"
                  >
                    code
                  </div>
                </div>
              </div>
            </div>
            <div
              class="d-inline-flex d-inline-flex d-xl-none px-2"
            >
              <div
                class="badge text-secondary border border-secondary rounded-0 position-relative text-uppercase miniBadge d-inline-flex d-xl-none px-2"
                data-testid="repo-kind-badge"
              >
                <div
                  class="d-flex flex-row align-items-center"
                >
                  code
                </div>
              </div>
            </div>
          </td>
          <td
            class="hoverableCell align-middle"
            role="button"
          >
            <div
              class="mx-auto px-2 text-center text-dark fw-bold badge"
            >
              85
            </div>
          </td>
          <td
            class="hoverableCell d-none d-md-table-cell align-middle"
            role="button"
          >
            <div
              class="mx-auto px-2 text-center text-dark fw-bold badge"
            >
              70
            </div>
          </td>
          <td
            class="hoverableCell d-none d-md-table-cell align-middle"
            role="button"
          >
            <div
              class="mx-auto px-2 text-center text-dark fw-bold badge"
            >
              100
            </div>
          </td>
          <td
            class="d-none d-md-table-cell align-middle"
          >
            <div
              class="mx-auto px-2 text-center"
            >
              <span
                class="text-muted"
              >
                n/a
              </span>
            </div>
          </td>
          <td
            class="d-none d-md-table-cell align-middle"
          >
            <div
              class="mx-auto px-2 text-center"
            >
              <span
                class="text-muted"
              >
                n/a
              </span>
            </div>
          </td>
          <td
            class="d-none d-md-table-cell align-middle"
          >
            <div
              class="mx-auto px-2 text-center"
            >
              <span
                class="text-muted"
              >
                n/a
              </span>
            </div>
          </td>
        </tr>
        <tr>
          <td
            class="align-middle repoCell darkBgCell"
          >
            <div
              class="d-flex flex-row align-items-center"
            >
              <button
                aria-label="Go from summary to section: spec"
                class="btn btn-link text-dark text-truncate fw-bold px-2 py-0 py-xl-1 repoBtn"
              >
                spec
              </button>
              <div
                class="d-inline-flex d-none d-xl-inline-flex"
              >
                <div
                  class="badge text-secondary border border-secondary rounded-0 position-relative text-uppercase miniBadge d-none d-xl-inline-flex"
                  data-testid="repo-kind-badge"
                >
                  <div
                    class="d-flex flex-row align-items-center"
                  >
                    community
                  </div>
                </div>
              </div>
            </div>
            <div
              class="d-inline-flex d-inline-flex d-xl-none px-2"
            >
              <div
                class="badge text-secondary border border-secondary rounded-0 position-relative text-uppercase miniBadge d-inline-flex d-xl-none px-2"
                data-testid="repo-kind-badge"
              >
                <div
                  class="d-flex flex-row align-items-center"
                >
                  community
                </div>
              </div>
            </div>
          </td>
          <td
            class="hoverableCell align-middle"
            role="button"
          >
            <div
              class="mx-auto px-2 text-center text-dark fw-bold badge"
            >
              44
            </div>
          </td>
          <td
            class="hoverableCell d-none d-md-table-cell align-middle"
            role="button"
          >
            <div
              class="mx-auto px-2 text-center text-dark fw-bold badge"
            >
              60
            </div>
          </td>
          <td
            class="hoverableCell d-none d-md-table-cell align-middle"
            role="button"
          >
            <div
              class="mx-auto px-2 text-center text-dark fw-bold badge"
            >
              80
            </div>
          </td>
          <td
            class="hoverableCell d-none d-md-table-cell align-middle"
            role="button"
          >
            <div
              class="mx-auto px-2 text-center text-dark fw-bold badge"
            >
              35
            </div>
          </td>
          <td
            class="hoverableCell d-none d-md-table-cell align-middle"
            role="button"
          >
            <div
              class="mx-auto px-2 text-center text-dark fw-bold badge"
            >
              0
            </div>
          </td>
          <td
            class="d-none d-md-table-cell align-middle"
          >
            <div
              class="mx-auto px-2 text-center"
            >
              <span
                class="text-muted"
              >
                n/a
              </span>
            </div>
          </td>
        </tr>
        <tr>
          <td
            class="align-middle repoCell darkBgCell"
          >
            <div
              class="d-flex flex-row align-items-center"
            >
              <button
                aria-label="Go from summary to section: sdk-java"
                class="btn btn-link text-dark text-truncate fw-bold px-2 py-0 py-xl-1 repoBtn"
              >
                sdk-java
              </button>
              <div
                class="d-inline-flex d-none d-xl-inline-flex"
              >
                <div
                  class="badge text-secondary border border-secondary rounded-0 position-relative text-uppercase miniBadge d-none d-xl-inline-flex"
                  data-testid="repo-kind-badge"
                >
                  <div
                    class="d-flex flex-row align-items-center"
                  >
                    code
                  </div>
                </div>
              </div>
            </div>
            <div
              class="d-inline-flex d-inline-flex d-xl-none px-2"
            >
              <div
                class="badge text-secondary border border-secondary rounded-0 position-relative text-uppercase miniBadge d-inline-flex d-xl-none px-2"
                data-testid="repo-kind-badge"
              >
                <div
                  class="d-flex flex-row align-items-center"
                >
                  code
                </div>
              </div>
            </div>
          </td>
          <td
            class="hoverableCell align-middle"
            role="button"
          >
            <div
              class="mx-auto px-2 text-center text-dark fw-bold badge"
            >
              85
            </div>
          </td>
          <td
            class="hoverableCell d-none d-md-table-cell align-middle"
            role="button"
          >
            <div
              class="mx-auto px-2 text-center text-dark fw-bold badge"
            >
              70
            </div>
          </td>
          <td
            class="hoverableCell d-none d-md-table-cell align-middle"
            role="button"
          >
            <div
              class="mx-auto px-2 text-center text-dark fw-bold badge"
            >
              100
            </div>
          </td>
          <td
            class="d-none d-md-table-cell align-middle"
          >
            <div
              class="mx-auto px-2 text-center"
            >
              <span
                class="text-muted"
              >
                n/a
              </span>
            </div>
          </td>
          <td
            class="d-none d-md-table-cell align-middle"
          >
            <div
              class="mx-auto px-2 text-center"
            >
              <span
                class="text-muted"
              >
                n/a
              </span>
            </div>
          </td>
          <td
            class="d-none d-md-table-cell align-middle"
          >
            <div
              class="mx-auto px-2 text-center"
            >
              <span
                class="text-muted"
              >
                n/a
              </span>
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
